R/nlm_mosaicfield.R

```diff
@@ -57,142 +56,147 @@ nlm_mosaicfield <- function(ncol,
                             collect     = FALSE,
                             infinit     = FALSE,
                             rescale     = TRUE) {
-  # Check function arguments ----
-  checkmate::assert_count(ncol, positive = TRUE)
-  checkmate::assert_count(nrow, positive = TRUE)
-  checkmate::assert_numeric(resolution)
-  checkmate::assert_count(n, positive = TRUE, na.ok = TRUE)
-  checkmate::assert_numeric(mosaic_mean)
-  checkmate::assert_numeric(mosaic_sd)
-  checkmate::assert_logical(collect)
-  checkmate::assert_logical(infinit)
-  checkmate::assert_logical(rescale)
-
-  mosaicfields_return <- list()
-
-  if (!is.na(n)) {
-    mosaicfield_result <- spatstat.random::rMosaicField(
-      spatstat.random::rpoislinetess(4),
-      stats::rnorm,
-      dimyx = c(nrow, ncol),
-      rgenargs = list(mean = mosaic_mean,
-                      sd = mosaic_sd)
-    )
-
-    if (isTRUE(collect)) {
-      mosaicfield_list <- list()
-      mosaicfield_list[[1]] <- mosaicfield_result
-    }
 
-    for (i in 2:n) {
-      mosaicfield_n <- spatstat.random::rMosaicField(
+  if (requireNamespace("RandomFieldsUtils", quietly = TRUE)) {
```

Comment:
We should check here for `RandomFields`, not `RandomFieldsUtils`, right?

Reply:
No, because it's only using RandomFieldsUtils not RandomFields.

Reply:
Ahhh, I see. Thanks :-)